### PR TITLE
Functional React Native example

### PR DESCRIPTION
**Pull request by alex-hladun**
## Describe the problem you have without this PR
* Outdated React-Native / Expo support. Running the tutorial did not work due to outdated Expo SDK. 
* Convert class-based component to functional and separate db initialization to separate file.
* React native package was referenced with hard-coded SDK. Use actual version.
* RXDB support for 4.0 added
* Update ReadMe with new environment setup link
* Implement native react context
* Add 'delete hero' functionality

### CODE REVIEW

File: examples/react-native/package.json [hunk 1]
```diff
@@ -24,14 +24,15 @@
   },
   "main": "node_modules/expo/AppEntry.js",
   "dependencies": {
-    "expo": "37.0.12",
+    "expo": "^43.0.2",
```
Comment (pubkey) [inline examples/react-native/package.json:27]: Please pin the version. We do not use package-lock for several reasons. So remove the `^`

File: examples/react-native/package.json [hunk 2]
```diff
@@ -24,14 +24,15 @@
...
     "global": "4.4.0",
     "pouchdb-adapter-asyncstorage": "6.4.1",
     "pouchdb-adapter-http": "7.2.2",
     "random-token": "0.0.8",
     "react": "17.0.2",
-    "react-native": "https://github.com/expo/react-native/archive/sdk-38.0.2.tar.gz",
+    "react-native": "0.64.2",
     "rxdb": "file:rxdb-local.tgz",
-    "rxjs": "7.3.1"
+    "rxdb-hooks": "^4.0.0",
```
Comment (pubkey) [inline examples/react-native/package.json:35]: Also pin the version here and at the line below.

File: examples/react-native/package.json (hunk 1 above)
Comment (alex-hladun) [reply examples/react-native/package.json:27]: Will do. Still in draft while I sort out some jest/react-native issues

**Comment by alex-hladun** [comment]
@pubkey Do you have any reccomendations on the failing tests? 
* react-native test fails the rxdb integrity check with a different checksum. I didn't touch the rxdb version.
* test-browser-couch test fails due to what looks like an unrelated mocha error

Not a testing expert.

**Comment by pubkey** [comment]
Ignore `test-browser-couch`. It is failing at master atm, likely because github changed something in github actions which has now slower servers. I will fix this when I have time.

I am not sure about the checksum missmatch.
I would be ok with removing the yarn lock, or better completely removing yarn if we do not need it anymore.

**Comment by pubkey** [comment]
@alex-hladun do we really have to add `rxdb-hooks`? The problem I see is when RxDB gets breaking updates, the CI will fail until rxdb-hooks is also updated.

**Comment by alex-hladun** [comment]
> @alex-hladun do we really have to add `rxdb-hooks`? The problem I see is when RxDB gets breaking updates, the CI will fail until rxdb-hooks is also updated.

No, our team just uses it as part of our architecture. I'll take it out.

**Comment by alex-hladun** [comment]
@pubkey Implemented native context and removed rxdb-hooks. Added delete hero function.

**Comment by pubkey** [comment]
Can this be merged? @alex-hladun

**Comment by alex-hladun** [comment]
Yep, everything is good on my end.
________________________________

**Pull request merged**